ml-agents/mlagents/trainers/settings.py

```diff
@@ -681,7 +681,12 @@ def structure(d: Mapping, t: type) -> Any:
 
     class DefaultTrainerDict(collections.defaultdict):
         def __init__(self, *args):
-            super().__init__(TrainerSettings, *args)
+            # Depending on how this is called, args may have the defaultdict
+            # callable at the start of the list or not.
```

Comment:
Maybe add a note about how it addresses an issue with pickle load for posterity

Reply:
Done